assets/scripts/app/ScrollIndicators.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { registerKeypress } from './keypress'
+
+class ScrollIndicators extends React.Component {
+  static propTypes = {
+    posLeft: PropTypes.number.isRequired,
+    posRight: PropTypes.number.isRequired,
+    scrollStreet: PropTypes.func.isRequired,
+    scrollTop: PropTypes.number.isRequired
+  }
+
+  shouldComponentUpdate (nextProps) {
+    if (this.props.posLeft !== nextProps.posLeft &&
+        this.props.posRight !== nextProps.posRight) {
+      this.refs.left_indicator.innerHTML = Array(nextProps.posLeft + 1).join('‹')
+      this.refs.right_indicator.innerHTML = Array(nextProps.posRight + 1).join('›')
+      return true
+    }
+    // Checking if window was resized
+    if (this.props.scrollTop !== nextProps.scrollTop) {
+      this.updateIndicatorsPosition(nextProps.scrollTop)
+      return true
+    }
```

Comment:
Same here, actually -- we can do `updateIndicatorsPosition()` in the `render()`! More details below.